Yoast/wordpress-seo
# Frontend robots presenter

## Summary


This PR can be summarized in the following changelog entry:

* Adds indexables support for the robots meta tag for simple posts.

## Relevant technical choices:

* Added `Indexable` mock class.

## Test instructions

This PR can be tested by following these steps:

* Make sure all robots settings are set to default:
  * `Settings` > `Reading` > `Search Engine Visibility` > Unchecked
  * `SEO` > `Search Appearance` > `Content Types` > Set `Show *** in search results` to Yes for all post types. 

### Default settings
* Make a single post or page for testing and publish it.
  * Make sure the advanced settings are:
    * `Allow search engines to show this Post in search results?` > Default: Yes
    * `Should search engines follow links on this Post?` > Yes
    * `Meta robots advanced` > `Site-wide default: None`
    * `Canonical URL` > Empty
  * Publish and inspect the post/page on the frontend and look for the `<meta property="robots" .../>` tag within the **FIRST** `This site is optimized with the Yoast SEO plugin` block. It should not be there, because when the value is `index,follow` it does not need to be shown.

### Post specific overrides
* Edit your post.
* Change the following settings:
  * `Allow search engines to show this Post in search results?` > `No`
* Publish and inspect the post/page on the frontend and look for the `<meta property="robots" .../>` tag within the **FIRST** `This site is optimized with the Yoast SEO plugin` block. The value should be `noindex,follow`.

* Edit your post.
* Change the following settings:
  * `Allow search engines to show this Post in search results?` > `Default: Yes`
  * `Should search engines follow links on this Post?` > No
* Publish and inspect the post/page on the frontend and look for the `<meta property="robots" .../>` tag within the **FIRST** `This site is optimized with the Yoast SEO plugin` block. The value should be `index,nofollow`.

* Edit your post.
* Change the following settings:
  * `Should search engines follow links on this Post?` > Yes
  * `Meta robots advanced`: `No Image Index`, `No Archive`, `No Snippet`
* Publish and inspect the post/page on the frontend and look for the `<meta property="robots" .../>` tag within the **FIRST** `This site is optimized with the Yoast SEO plugin` block. The value should be `noimageindex,noarchive,nosnippet`. Notice also that `index,follow` have been left out, because they are defaults.

After going through all these steps set all settings back to their defaults.

### Site wide overrides

* Change the following settings in your admin:
  * `SEO` > `Search Appearance` > `Content Types` > `Posts` > `Show Posts in search results?` > `No`
  * `SEO` > `Search Appearance` > `Content Types` > `Pages` > `Show Posts in search results?` > `No`
(Use the yoast test helper for this custom post type)
  * `SEO` > `Search Appearance` > `Content Types` > `Books` > `Show Posts in search results?` > `No`
* Publish and inspect the post/page/book on the frontend and look for the `<meta property="robots" .../>` tag within the **FIRST** `This site is optimized with the Yoast SEO plugin` block. The value should be `noindex,follow`.
* Revert the settings.

* Change the following settings in your admin:
  * `Settings` > `Reading` > `Search Engine Visibility` > Checked
* Publish and inspect the post/page on the frontend and look for the `<meta property="robots" .../>` tag within the **FIRST** `This site is optimized with the Yoast SEO plugin` block. The value should be `noindex,follow`.

## UI changes

## Documentation

## Quality assurance


Fixes #
